sampler.go

```diff
@@ -252,7 +266,7 @@ func (s *GuaranteedThroughputProbabilisticSampler) Equal(other Sampler) bool {
 }
 
 // this function should only be called while holding a Write lock
-func (s *GuaranteedThroughputProbabilisticSampler) update(lowerBound, samplingRate float64) error {
+func (s *GuaranteedThroughputProbabilisticSampler) update(lowerBound, samplingRate, upperBound float64) error {
 	if s.samplingRate != samplingRate {
 		tags := []Tag{
```

Comment:
another example. We should consolidate all updates into small functions, rather than repeating code that can potentially diverge.